include/znc/Modules.h

```diff
@@ -603,6 +603,16 @@ class CModule {
                          CChan& Channel, bool bNoChange);
     virtual void OnDeop(const CNick& OpNick, const CNick& Nick, CChan& Channel,
                         bool bNoChange);
+    /** Called when a nick is halfopped on a channel */
+    virtual void OnHalfOp2(const CNick* pOpNick, const CNick& Nick, CChan& Channel,
+                       bool bNoChange);
+    virtual void OnHalfOp(const CNick& OpNick, const CNick& Nick, CChan& Channel,
```

Comment:
No need for this "2". The other function has it only for back compatibility with old modules.